templates/cluster/gcp-hosted-cp/templates/gcpcluster.yaml

```diff
@@ -0,0 +1,18 @@
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: GCPCluster
+metadata:
+  name: {{ include "cluster.name" . }}
+  finalizers:
+    - k0rdent.mirantis.com/cleanup
```

Comment:
out of curiosity: is the finalizer actually required? are there any orphans after a `gcpcluster` object removal?

Reply:
I didn’t encounter any leftover resources after removing the GCPCluster, but I decided to keep this finalizer as we do for other providers. Just as a precaution in case of any unnoticed issues. Since the controller already includes the necessary cleanup logic, no additional code changes are required.

Reply:
ok got it thanks